transactions/WFLY-15659-Transaction-SlotStore-config-options.adoc

```diff
@@ -0,0 +1,118 @@
+
+== Issue Metadata
+
+=== Issue
+
+* https://issues.redhat.com/browse/WFLY-15659
+
+=== Related Issues
+
+* https://issues.redhat.com/browse/JBTM-3276
+* https://issues.redhat.com/browse/JBTM-3417
+
+=== Dev Contacts
+
+* mailto:{email}[{author}]
+
+=== QE Contacts
+
+=== Testing By
+// Put an x in the relevant field to indicate if testing will be done by Engineering or QE. 
+// Discuss with QE during the Kickoff state to decide this
+* [ ] Engineering
+
+* [ ] QE
+
+=== Affected Projects or Components
+
+Wildfly
+
+=== Other Interested Projects
+
+Narayana, Infinispan, Mashona.
+
+=== Relevant Installation Types
+// Remove the x next to the relevant field if the feature in question is not relevant
+// to that kind of WildFly installation
+* [x] Traditional standalone server (unzipped or provisioned by Galleon)
+
+* [x] Managed domain
+
+* [x] OpenShift s2i
+
+* [x] Bootable jar
+
+== Requirements
+
+=== Hard Requirements
+
+Extend the server management model to facilitate configuration of the new SlotStore transaction log type.
+
+=== Nice-to-Have Requirements
+
+Extend the sever dependency model to allow use of the Persistent Memory library, mashona, to support SlotStore use on pmem hardware. Optionally the components consuming it could simply bundle their own copies, trading version flexibility vs. footprint.
+
+=== Non-Requirements
+
+== Backwards Compatibility
+
+Existing store types remain available, existing configurations remain valid.
+Automatic data migration between store types is not supported, so no transactions should be in-flight when switching store types.
+
+=== Default Configuration
+
+The existing transaction store will remain the default. The SlotStore will be available only by explicit user selection. The SlotStore will provide defaults for its configuration properties where possible.
+
+=== Importing Existing Configuration
+
+=== Deployments
+
+=== Interoperability
+
+== Implementation Plan
+
+This will be a community-only feature for a few releases before being considered for support in EAP.
+
+== Security Considerations
+
+As with existing transaction store types the tx log files may, in certain cases, contain db connection information.
+
+////
+Identification if any security implications that may need to be considered with this feature
+or a confirmation that there are no security implications to consider.
+////
+
+== Test Plan
+
+* Testing of the new server configuration options will require new tests, patterned on those for existing store configurations.
+
+* Testing of the SlotStore itself can be accomplished by using the same transaction tests that exercise existing store types, but changing the server config to use the new store type.
+
+* Testing of the pmem options will require appropriate hardware, though this can be simulated by system configuration (similar to a RAM disk)
```

Comment:
How much of this is covered within Narayana testing?

Reply:
The ObjectStore interface - all of it. The SlotStore implementation of it - no idea. Theoretically all of it, but that would require running all the store tests against each store implementation, which makes for quite a large matrix. I don't know what the current CI setup does. pmem - none, as the tx CI doesn't have any. I run the mashona tests on real pmem hardware for each release, but don't run the tx test suite, though that should be possible. Feels like if the CI for that has  to be somewhere, it's better on the narayana side using fake-pmem or just the SlotStore on SSD, rather than on the mashona side. The advantage of having real pmem hardware is in accurate perf numbers for e.g. regression testing, not in functional testing.